### ISSUE TITLE
Sub-flows features

### ISSUE THREAD

**Issue by info-rchitect** [Origen-SDK/o2#74]
Hi,

This is an issue regarding sub-flows features.  Here are a couple items that have come up:

1. Should we continue to reload the target by default as in O1?
2. When a sub-flow is imported multiple times, users would like to be able to name the generated groups using options

Feel free to add on.

**Comment by redxeth** [comment]
Good idea-- also some additional features that we have needed at various times in O1 related to sub-flows (maybe we're not meaning the same kinds of sub-flows-- but will throw this out there):

- Easy way to get pattern name in the flow itself
- Easy way to get flow or sub-flow file name in flow itself-- this would be handy so when later looking at a generated test flow you can FIND where the test was defined, esp if you have a complex flow/sub-flow structure.
- For a sub-flow to access the flow (or other sub-flow) calling it.
